### ISSUE TITLE
dnn.dnn_conv3d but filter_dilation is only 3D

### ISSUE THREAD

**Issue by trungnt13** [Theano/Theano#4680]
I think the new implementation of filter_dilation messed up with dnn_conv3d.
This is the full traceback (subsample is 3D but filter_dilation is only 2D):

```
Traceback (most recent call last):
  File "/Users/trungnt13/tmp/tmp6.py", line 18, in <module>
    y = f1(x)
  File "/Users/trungnt13/libs/odin/odin/nnet/base.py", line 235, in __call__
    return self.apply(*args, **kwargs)
  File "/Users/trungnt13/libs/odin/odin/nnet/base.py", line 224, in apply
    out = self._apply(*args, **kwargs)
  File "/Users/trungnt13/libs/odin/odin/nnet/conv.py", line 159, in _apply
    conved = self.convolve(x)
  File "/Users/trungnt13/libs/odin/odin/nnet/conv.py", line 545, in convolve
    filter_shape=K.get_shape(self.W))
  File "/Users/trungnt13/libs/odin/odin/backend/theano/tensor.py", line 1364, in conv3d
    conv_mode='conv')
  File "/Users/trungnt13/libs/conda/envs/ai/lib/python2.7/site-packages/theano/sandbox/cuda/dnn.py", line 1283, in dnn_conv3d
    desc_op.subsample)
  File "/Users/trungnt13/libs/conda/envs/ai/lib/python2.7/site-packages/theano/sandbox/cuda/dnn.py", line 600, in get_out_shape
    subsample)
  File "/Users/trungnt13/libs/conda/envs/ai/lib/python2.7/site-packages/theano/tensor/nnet/abstract_conv.py", line 76, in get_conv_output_shape
    subsample[i], filter_dilation[i]) for i in range(len(subsample)))
  File "/Users/trungnt13/libs/conda/envs/ai/lib/python2.7/site-packages/theano/tensor/nnet/abstract_conv.py", line 76, in <genexpr>
    subsample[i], filter_dilation[i]) for i in range(len(subsample)))
IndexError: tuple index out of range
```

**Comment by Sentient07** [comment]
This issue has been addressed in #4570 . To be a little more precise, this is a part of this [commit](https://github.com/Theano/Theano/pull/4570/commits/3e71f63a1a07cc1175f8e143246644e873682a83)

**Comment by trungnt13** [comment]
```
Traceback (most recent call last):
  File "/Users/trungnt13/tmp/tmp3.py", line 19, in <module>
    print(K.eval(f(X)).shape)
  File "/Users/trungnt13/libs/odin/odin/nnet/base.py", line 263, in __call__
    return self.apply(*args, **kwargs)
  File "/Users/trungnt13/libs/odin/odin/nnet/base.py", line 258, in apply
    out = self._apply(*args, **kwargs)
  File "/Users/trungnt13/libs/odin/odin/nnet/conv.py", line 158, in _apply
    conved = self.convolve(x)
  File "/Users/trungnt13/libs/odin/odin/nnet/conv.py", line 544, in convolve
    filter_shape=K.get_shape(self.W))
  File "/Users/trungnt13/libs/odin/odin/backend/theano/tensor.py", line 1382, in conv3d
    filters_shape=None)
  File "/Users/trungnt13/libs/conda/envs/ai/lib/python2.7/site-packages/theano/tensor/nnet/conv3d2d.py", line 251, in conv3d
    border_mode=border_mode[1])  # ignoring border_mode[2]
  File "/Users/trungnt13/libs/conda/envs/ai/lib/python2.7/site-packages/theano/tensor/nnet/__init__.py", line 148, in conv2d
    filter_dilation)
  File "/Users/trungnt13/libs/conda/envs/ai/lib/python2.7/site-packages/theano/tensor/nnet/abstract_conv.py", line 162, in conv2d
    filter_dilation=filter_dilation)
  File "/Users/trungnt13/libs/conda/envs/ai/lib/python2.7/site-packages/theano/tensor/nnet/abstract_conv.py", line 795, in __init__
    filter_dilation)
  File "/Users/trungnt13/libs/conda/envs/ai/lib/python2.7/site-packages/theano/tensor/nnet/abstract_conv.py", line 676, in __init__
    pad_h, pad_w = map(int, border_mode)
```

I check the newest commit #e4e0878, this issue still there.
